flink-runtime/src/main/java/org/apache/flink/runtime/state/StateBackendBuilder.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.state;
+
+import java.io.IOException;
+import java.util.Collection;
+
+/**
+ * A mutable builder to build a state backend instance
+ *
+ * @param <T> The type of the state backend instance
+ * @param <S> The type of the state handle
+ */
+public interface StateBackendBuilder<T, S> {
+	T build() throws IOException;
+
+	StateBackendBuilder setRestoreStateHandles(Collection<S> restoreStateHandles);
```

Comment:
I think this is not required to be part of the interface. When I look at the usage, the state handles could simply be supplied to the constructor of the `StateBackendBuilder` implementation. This make sense if there are backends in the future that might restore differently than with state handles (e.g. one of those you are working on). Furthermore, then I wonder if this is even needed or we could just `SupplierWithException` as the root interface.

Reply:
Let me make `stateHandles` parameter in builder constructor and remove the setter from interface. And I agree that we could just use `SupplierWithException` but this will force the method name to be `get` instead of `build` and make it hard to detect the caller of builder. So I suggest to still create the builder interface. What's your opinion?